### PR TITLE
Add syntax check before validation

### PR DESCRIPTION
**Pull request by Askanio234**
Also added tests for has_no_variables_that_shadow_default_names validator, put them in general group, because they do not mix well with syntax error fixtures.

### CODE REVIEW

File: fiasko_bro/code_validator.py [hunk 1]
```diff
@@ -158,4 +167,10 @@ def validate(self, repo_path, original_repo_path=None, check_repo_size=True, **k
         self.validator_arguments['solution_repo'] = LocalRepositoryInfo(repo_path)
         if original_repo_path:
             self.validator_arguments['original_repo'] = LocalRepositoryInfo(original_repo_path)
+        syntax_errors = self.get_syntax_errors_in_repos(
+            self.validator_arguments['solution_repo'],
+            self.validator_arguments.get('original_repo', None)
```
Comment (patrnk) [inline fiasko_bro/code_validator.py]: `get` returns `None` anyway, no need to indicate that.

File: fiasko_bro/code_validator.py [hunk 2]
```diff
@@ -146,6 +146,15 @@ def run_validator_group(self, group, add_warnings=False, *args, **kwargs):
             return errors
         return errors
 
+    def get_syntax_errors_in_repos(self, solution_repo, original_repo=None):
+        syntax_errors = validators.has_no_syntax_errors(solution_repo)
+        if syntax_errors:
+            return [syntax_errors]
+        if original_repo:
```
Comment (patrnk) [inline fiasko_bro/code_validator.py]: Actually, we don't care if the `original_repo` contains any syntax errors, we never use its syntax tree (and we don't throw the exception while analyzing it: https://github.com/devmanorg/fiasko_bro/blob/master/fiasko_bro/repository_info.py#L53).

It's actually a mistake that we create the tree for the original repo at all. We are going to fix it in `0.0.3`, but for now, let's just ignore it if `original_repo` contains any syntax errors.

File: fiasko_bro/code_validator.py [hunk 3]
```diff
@@ -146,6 +146,11 @@ def run_validator_group(self, group, add_warnings=False, *args, **kwargs):
             return errors
         return errors
 
+    def get_syntax_errors_in_repo(self, repo):
```
Comment (patrnk) [inline fiasko_bro/code_validator.py]: Now you can simply get rid of the method. Take a closer look at the code in `validate`: doesn't it do the same thing?

**Comment by Melevir** [comment]
Why can't we user pre validator checks for that?

**Comment by patrnk** [comment]
We shouldn't use pre validator checks for that, that would break warning validators.

The issue is due to a bug in code validator. Please don't merge this PR, I will submit the fix shortly

**Comment by Melevir** [comment]
I won't merge it anyway since I don't understand why we can't use prevalidators for that.

**Comment by Askanio234** [comment]
Well we can't use prevalidators, because prevalidators run before LocalRepositoryInfo object is created, and we check for syntax errors during creation of LocalRepositoryInfo object, currently everything fails if there is indeed a syntax error in repo see #80.

**Comment by patrnk** [comment]
In #85, I tried to show that there's no need to move the syntax validator anywhere. The underlying bug is in the `validate` method. I'm sorry I haven't spotted that earlier.

**Pull request closed without merging**